### PR TITLE
feat: add support for private locations

### LINKED ISSUE #159: story: ad support for private locations
### Allow provider to handle new Checkly Private Locations

- [ ] Add CRUD methods
- [ ] Allow private location assignments in check/groups
- [ ] Add the required tests
- [ ] Update docs and examples

### PR DESCRIPTION
**Pull request by pilimartinez**
## Affected Components
* [X] Resources
* [X] Tests
* [X] Docs
* [ ] Tooling
* [ ] Other

## Pre-Requisites
* [X] Terraform code is formatted with `terraform fmt`
* [X] Go code is formatted with `go fmt`
* [X] `plan` & `apply` command of `demo/main.tf` file do not produce diffs

## Notes for the Reviewer

- Add CRUD methods
- New tests
- Update docs and examples

> Resolves #159

### CODE REVIEW

File: checkly/provider.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func Provider() *schema.Provider {
 			"checkly_trigger_check":        resourceTriggerCheck(),
 			"checkly_trigger_group":        resourceTriggerGroup(),
 			"checkly_environment_variable": resourceEnvironmentVariable(),
+			"checkly_private_locations":    resourcePrivateLocations(),
```
Comment (ianaya89) [inline checkly/provider.go:43]: This should be singular "location"

File: checkly/resource_private_locations.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	checkly "github.com/checkly/checkly-go-sdk"
+)
+
+func resourcePrivateLocations() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePrivateLocationsCreate,
```
Comment (ianaya89) [inline checkly/resource_private_locations.go:15]: Same for this method names

File: checkly/resource_private_locations.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		return fmt.Errorf("resourcePrivateLocationsCreate: translation error: %w", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), apiCallTimeout())
+	defer cancel()
+	result, err := client.(checkly.Client).CreatePrivateLocation(ctx, pl)
+	if err != nil {
+		return fmt.Errorf("CreatePrivateLocation: API error: %w", err)
+	}
+	d.SetId(result.ID)
+	d.Set("raw_key", result.Keys[0].RawKey)
```
Comment (ianaya89) [inline checkly/resource_private_locations.go:65]: I think will be better to name this property `key`

File: docs/resources/private_locations.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (ianaya89) [inline docs/resources/private_locations.md:1]: Did you create this with the docs script or was manually added?

File: checkly/resource_private_locations.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	return &schema.Resource{
+		Create: resourcePrivateLocationsCreate,
+		Read:   resourcePrivateLocationsRead,
+		Update: resourcePrivateLocationsUpdate,
+		Delete: resourcePrivateLocationsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
```
Comment (ianaya89) [inline checkly/resource_private_locations.go:23]: All these fields need to have the `Description` property

**Review by ianaya89 — CHANGES_REQUESTED** [review]

File: checkly/integration_test.go [hunk 6]
```diff
@@ -1,3 +1,4 @@
+//go:build integration
```
[the remark this answers is not in the document]
Comment (pilimartinez) [reply checkly/integration_test.go:1]: It appears every time I save, I think it comes from the // +build integration thats on the next line. That came from the main branch and that's why I left it there but if its not necessary I can remove it (and that will prevent the //go:build integration to appear on the code)

**Review by ianaya89 — DISMISSED** [review]

**Pull request merged**